godot-core/src/meta/property_update.rs

```diff
@@ -0,0 +1,14 @@
+pub struct PropertyUpdate<'a, C, T> {
+    pub new_value: T,
+    pub field_name: &'a str, // might also be &'a StringName, depending on what's available
+    pub get_field_mut: fn(&mut C) -> &mut T,
+}
```

Comment:
Re public fields, I mentioned something [here](https://github.com/godot-rust/gdext/pull/946#issuecomment-2495624782):

> I'm not sure how I feel about public fields; it takes away our freedoms of changing implementation, optimizing things, etc. At least the struct should be `#[non_exhaustive]` to allow field additions, but maybe getters are better. Which would mean we can't use destructuring in case of (A).

What's your take on this?